src/main/java/net/fhirfactory/pegacorn/fhir/operations/OperationOutcomeGenerator.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2020 Mark A. Hunter
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fhirfactory.pegacorn.fhir.operations;
+
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.OperationOutcome;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class OperationOutcomeGenerator {
+
+    public OperationOutcome generateResourceGetOutcome(IdType resourceId, OperationOutcomeSeverityEnum severityEnum){
+        // Create Issue
+        OperationOutcome.OperationOutcomeIssueComponent issue = new OperationOutcome.OperationOutcomeIssueComponent();
+        issue.setSeverity(OperationOutcome.IssueSeverity.valueOf(severityEnum.getSeverityCode()));
+        issue.setCode(OperationOutcome.IssueType.INFORMATIONAL);
+        CodeableConcept details = new CodeableConcept();
+        Coding detailsCoding = new Coding();
+        detailsCoding.setCode("MSG_RETRIEVED");
+        detailsCoding.setDisplay("Resource Id " + resourceId + " retrieved");
+        detailsCoding.setSystem("http://hl7.org/fhir/ValueSet/operation-outcome");
+        detailsCoding.setVersion("4.0.1");
+        details.addCoding(detailsCoding);
+        details.setText("Resource Id " + resourceId + " retrieved");
+        issue.setDetails(details);
+        // Create Outcome
+        OperationOutcome outcome = new OperationOutcome();
+        outcome.setId(resourceId);
```

Comment:
Is it intentional that resourceId isn't checked like it is in the other methods?  Should all the methods call a local method to centralise the logic? e.g.
```
    public OperationOutcome generateResourceGetOutcome(IdType resourceId, OperationOutcomeSeverityEnum severityEnum){
        OperationOutcome outcome = generateResourceUpdateFailureOutcome("MSG_RETRIEVED", "Resource Id " + resourceId + " retrieved", 
                resourceId, severityEnum, IssueType.INFORMATIONAL);
        return(outcome);
    }

    public OperationOutcome generateResourceNotFoundOutcome(IdType resourceId, OperationOutcomeSeverityEnum severityEnum){
        OperationOutcome outcome = generateResourceUpdateFailureOutcome("MSG_NO_EXIST", "Resource Id " + resourceId + " does not exist", 
                resourceId, severityEnum, IssueType.NOTFOUND);
        return(outcome);
    }

    public OperationOutcome generateResourceDeletedOutcome(IdType resourceId, OperationOutcomeSeverityEnum severityEnum){
        OperationOutcome outcome = generateResourceUpdateFailureOutcome("MSG_DELETED", "The resource has been deleted", 
                resourceId, severityEnum, IssueType.DELETED);
        return(outcome);
    }

    public OperationOutcome generateResourceDeleteFailedOutcome(IdType resourceId, OperationOutcomeSeverityEnum severityEnum){
        OperationOutcome outcome = generateResourceUpdateFailureOutcome("MSG_AUTH_REQUIRED", "You must authenticate before you can use this service", 
                resourceId, severityEnum, IssueType.EXCEPTION);
        return(outcome);
    }


    public OperationOutcome generateResourceAddedOutcome(IdType resourceId, OperationOutcomeSeverityEnum severityEnum){
        OperationOutcome outcome = generateResourceUpdateFailureOutcome("MSG_CREATED", "New resource created", 
                resourceId, severityEnum, IssueType.INFORMATIONAL);
        return(outcome);
    }

    public OperationOutcome generateResourceFailedAddOutcome(IdType resourceId, OperationOutcomeSeverityEnum severityEnum){
        OperationOutcome outcome = generateResourceUpdateFailureOutcome("MSG_BAD_SYNTAX", "Bad Syntax in " + resourceId, 
                resourceId, severityEnum, IssueType.EXCEPTION);
        return(outcome);
    }

    public OperationOutcome generateResourceUpdateOutcome(IdType resourceId, OperationOutcomeSeverityEnum severityEnum){
        OperationOutcome outcome = generateResourceUpdateFailureOutcome("MSG_UPDATED", "existing resource updated", 
                resourceId, severityEnum, IssueType.INFORMATIONAL);
        return(outcome);
    }

    public OperationOutcome generateResourceUpdateFailureOutcome(IdType resourceId, OperationOutcomeSeverityEnum severityEnum){
        OperationOutcome outcome = generateResourceUpdateFailureOutcome("MSG_INVALID_ID", "Id not accepted", 
                resourceId, severityEnum, IssueType.EXCEPTION);
        return(outcome);
    }

    public OperationOutcome generateEmptyOutcome(){
        OperationOutcome outcome = generateResourceUpdateFailureOutcome("FAILED_OPERATION", "Failed Operation", 
                null, OperationOutcomeSeverityEnum.SEVERITY_ERROR, IssueType.EXCEPTION);
        return(outcome);
    }

    protected OperationOutcome generateResourceUpdateFailureOutcome(String code, String msg, IdType resourceId, OperationOutcomeSeverityEnum severityEnum, IssueType issueType) {
        // Create Issue
        OperationOutcome.OperationOutcomeIssueComponent issue = new OperationOutcome.OperationOutcomeIssueComponent();
        issue.setSeverity(OperationOutcome.IssueSeverity.valueOf(severityEnum.getSeverityCode()));
        issue.setCode(issueType);
        CodeableConcept details = new CodeableConcept();
        Coding detailsCoding = new Coding();
        detailsCoding.setCode(code);
        detailsCoding.setDisplay(msg);
        detailsCoding.setSystem("http://hl7.org/fhir/ValueSet/operation-outcome");
        detailsCoding.setVersion("4.0.1");
        details.addCoding(detailsCoding);
        details.setText(msg);
        issue.setDetails(details);
        // Create Outcome
        OperationOutcome outcome = new OperationOutcome();
        if(resourceId != null) {
            outcome.setId(resourceId);
        } else {
            outcome.setId("Unknown Id");
        }
        outcome.addIssue(issue);
        return(outcome);
    }
```